baseplate/ratelimit.py

```diff
@@ -0,0 +1,114 @@
+"""Simple bucketed rate limits in memcached or redis.
+
+Break up time into fixed buckets and allow a maximum number of counts per
+buckets.
+
+Buckets are formed by dividing the 86400 seconds of the day into equal length
+intervals.
+
+Example:
+
+If an interval length of 60 seconds is used there are 1440 buckets, one for
+each minute. When the minute ticks over (e.g. from 12:31:59 to 12:32:00) the
+count is reset because we use a new bucket.
+
+"""
+
+from __future__ import division
+
+import datetime
+import pytz
+
+from pymemcache.client.base import PooledClient
+from redis import StrictRedis
+
+
+class WrappedMemcache(object):
+    def __init__(self, cache):
+        self.cache = cache
+
+    def incr_and_get(self, key, ttl):
+        self.cache.add(key, 0, expire=ttl)
+        count = self.cache.incr(key, 1)
+        return count
+
+
+class WrappedRedis(object):
+    def __init__(self, cache):
+        self.cache = cache
+
+    def incr_and_get(self, key, ttl):
+        with self.cache.pipeline('incr') as pipe:
+            pipe.incr(key, 1)
+            pipe.expire(key, time=ttl)
+            responses = pipe.execute()
+
+        # extract the first response from the pipeline, which is the value of
+        # the key that was incr'd
+        count = responses[0]
+        return count
+
+
+def get_current_interval(interval_seconds):
+    """Convert the current time to an interval.
+
+    The 86400 seconds in a day are broken into into buckets of length
+    ``interval_seconds``. The current interval is then the bucket that the
+    current time falls into.
+
+    """
+    assert interval_seconds <= 86400, 'maximum interval is 1 day'
+
+    now = datetime.datetime.now(pytz.UTC)
+    seconds = (now.hour * 3600) + (now.minute * 60) + now.second
```

Comment:
Why not just something like `time.time() // interval`?  Then we can drop the dep on pytz and maybe the restriction on interval size?